worker/src/RTC/TransportCongestionControlClient.cpp

```diff
@@ -163,9 +164,57 @@ namespace RTC
 	{
 		MS_TRACE();
 
+		// Update packet loss history
+		size_t expected_packets = feedback->GetPacketStatusCount();
+		size_t lost_packets = 0;
+		for (const auto& result : feedback->GetPacketResults())
+		{
+			if (!result.received)
+				lost_packets += 1;
+		}
+		this->UpdatePacketLoss(static_cast<double>(lost_packets) / expected_packets);
+
 		this->rtpTransportControllerSend->OnTransportFeedback(*feedback);
 	}
 
+	void TransportCongestionControlClient::UpdatePacketLoss(double packetLoss)
+	{
+		// Add the score into the histogram.
+		if (this->packetLossHistory.size() == PacketLossHistogramLength)
+			this->packetLossHistory.erase(this->packetLossHistory.begin());
+
+		this->packetLossHistory.push_back(packetLoss);
+
+		/*
+		 * Scoring mechanism is a weighted average.
+		 *
+		 * The more recent the score is, the more weight it has.
+		 * The oldest score has a weight of 1 and subsequent scores weight is
+		 * increased by one sequentially.
+		 *
+		 * Ie:
+		 * - scores: [1,2,3,4]
+		 * - this->scores = ((1) + (2+2) + (3+3+3) + (4+4+4+4)) / 10 = 2.8 => 3
+		 */
+
+		size_t weight{ 0 };
+		size_t samples{ 0 };
+		double totalPacketLoss{ 0 };
+
+		for (auto packetLossEntry : this->packetLossHistory)
+		{
+			weight++;
+			samples += weight;
+			totalPacketLoss += weight * packetLossEntry;
+		}
+
+		[[maybe_unused]] double previousPacketLoss = this->packetLoss;
```

Comment:
Mediasoup uses C++11 standard, but this is a C++17 feature, I don't think we can use it yet